### DISCUSSION TITLE
NES ld65 VECTORS

### DISCUSSION THREAD

**Discussion by maliDjuro** [cc65/cc65#2211] [Q&A] [answered]
Hello, I've got a problem with NES VECTORS segment in ld65 (there's C code in the whole package).
It wants VECTORS to be of size $0c, but...
When I had the package coded in assembly, the size was 6 and it put the right vector pointers in.
And now, when I have the C code in the package, it writes some vectors that don't point to my routines (any of the 6 vectors),
but probably from its default scripts.
I put my own config file with -C, it was working with asm, but with C included it seems partially ignored...
How can I put the right vector pointers in without any of the precoded ld65 scripts?
Thank you for your time.

**Comment by mrdudz** [comment]
Please give a minimal example

**Reply by maliDjuro** [reply]
I can't, because the project is huge 256k PRG on MMC3, that's why I wrote that I have a problem when including parts in C.
When I was doing the thing in assembly, there were no problems, but I had to abandon that because of bugs and people suggesting it'll be easier for others to help if at least the parts were in C.
The same config worked as it was supposed to, but with C, ld65 reported VECTORS is overflowing and with size $0c it works.
Now the only problem is that it doesn't put the correct vector pointers into VECTORS segment.

**Reply by maliDjuro** [reply]
I can only put the config file.
[mmc3.txt](https://github.com/cc65/cc65/files/12830785/mmc3.txt)

**Comment by mrdudz** [comment]
That config file looks broken - when you load segments into a memory region, they will be appended to each other. so in this case right after RODATA. the "start" attribute does NOT affect where in the output file the segment will go - which is why originally there is a seperate memory region for the vectors.

**Reply by maliDjuro** [reply]
Thank you very much for your time and your will.
I'll try it on monday and then write here.
Much obliged.

**Reply by maliDjuro** [reply]
I've tried it, but the linker still puts its own vectors, I'd like to know if anyone knows how to circumvent the linker scripts that are compiled in it.
Thank you very much.

**Reply by maliDjuro** [reply, marked as answer]
I've corrected the mmc3.cfg, vectors now work, so if anyone needs it for future reference:
[mmc3.txt](https://github.com/cc65/cc65/files/12845113/mmc3.txt)

And don't use .org in the assembly files where you include stuff, let the cfg place them where they're supposed to be.
